buildbot/master.cfg

```diff
@@ -532,7 +538,10 @@ def make_portwatcher_factory(triggerable):
         return {'subportlist': ' '.join(subports)}
 
     portwatcher_factory.addStep(SetPropertyFromCommandWithPortlist(
-        command=util.WithProperties('./mpbb/mpbb --prefix '+prefix+' list-subports --archive-site '+config['archivesite']+' %(fullportlist)s'),
+        command=['./mpbb/mpbb', '--prefix', util.WithProperties(prefix), 'list-subports',
+                 '--archive-site', config['archivesite'],
+                 '--archive-site-private', config['archivesiteprivate'],
```

Comment:
Just curious: why do we need two different flags here? Just for simplicity? I understand that we need different flags when uploading files (or checking which ones need to be uploaded), but for preparing a list of subports to build, it shouldn't make any difference to whether a site is public or private?

Reply:
I wanted to make sure that we include the port in the list if its archive is not on the *correct* server (not just if its archive is not on *any* server). Consider: somebody adds a new port and forgets to set the license, or sets the wrong license, and this makes the port nondistributable. Buildbot would build it and upload the archive to the private server. Then we fix the license in a later commit so that it becomes distributable. I want to make sure that, now that the port's distributability status has changed, the port will be included in the list again so that the archive will be uploaded to the public server.

Reply:
My question was more like: why not generally supporting multiple archive sites (whatever the syntax) without explicitly declaring one as public and the other one as private.

Are you saying that now the port would be rebuilt if present on the wrong archive site?

Reply:
Because mpbb needs to know which URL is for distributable archives and which URL is for nondistributable archives, so that:

* If the archive is distributable:
  * If the archive is not already on the public (i.e. distributable) server, list it
  * If the archive is already on the public server, don't list it
* If the archive is not distributable:
  * If the archive is not already on the private (i.e. nondistributable) server, list it
  * If the archive is already on the private server, don't list it

Listing a port does not necessarily mean that it will be recompiled, it just means that a portbuilder will be scheduled for it. In the case I mentioned above, where a port's distributability status changes without its version or revision changing, the archive is already present on the buildworker, so the portbuilder will not need to recompile it, but it will then be uploaded to the other server during the gathering and uploading steps.